### PR TITLE
cloud: rely only on cloud-init to determine cloud and instance IDs (SC-337)

### PR DESCRIPTION
**Pull request by renanrodrigo**
Since the Xenial release, cloud-init provides us `cloud-id` to determine on which cloud are we running, if any. We used to check for fallback files in the case cloud-id was not present. The same logic applied to instance ID.

The proposal here is that, for the future releases, this is not needed anymore.

## Desired commit type::
 - [ ] Squash and merge: Using the "Proposed Commit Message"
 - [ ] Create a merge commit and use "Proposed Commit Message"
 - [x] Rebase and merge, no merge commit, rely only on existing commit messages

## Checklist:
 - [x] I have updated or added any unit tests accordingly
 - [ ] I have updated or added any integration tests accordingly
 - [ ] I have updated or added any documentation accordingly

LP: #1940131

### CODE REVIEW

**Review by lucasmoura — APPROVED** [review]
I think this is good, but let's wait for a review from @blackboxsw to see if we are not missing anything here

**Review by blackboxsw — APPROVED** [review]
Thanks @renanrodrigo this works and thank you for clarifying in the commit message here on this so we can track details better in launchpad.

**Review by blackboxsw — APPROVED** [review]
Confirmed integration test failures are not present locally.
1 feature passed, 0 failed, 0 skipped
1 scenario passed, 0 failed, 39 skipped
25 steps passed, 0 failed, 247 skipped, 0 undefined
Took 4m11.394s
___________________________________ summary ____________________________________
  behave-lxd-16.04: commands succeeded
  congratulations :)

**Pull request merged**
